### PR TITLE
Update requires for purge in views

### PR DESCRIPTION
**Pull request by julienlancelot**

### CODE REVIEW

File: sonar-db/src/main/java/org/sonar/db/component/ComponentDao.java [hunk 1]
```diff
@@ -147,6 +148,7 @@ public boolean existsById(Long id, DbSession session) {
     return mapper(session).selectComponentsHavingSameKeyOrderedById(key);
   }
 
+
```
Comment (sns-seb) [inline sonar-db/src/main/java/org/sonar/db/component/ComponentDao.java]: useless line return

File: sonar-db/src/main/java/org/sonar/db/component/ComponentDao.java [hunk 2]
```diff
@@ -251,6 +253,16 @@ public long countGhostProjects(DbSession session, @Nullable String query) {
...
   }
 
+  /**
+   * Retrieve enabled components keys with given qualifiers
+   *
+   * Used by Views plugin
+   */
+  public List<ComponentDto> selectComponentsByQualifiers(DbSession dbSession, Collection<String> qualifiers) {
+    checkArgument(!qualifiers.isEmpty(), "Qualifiers cannot be empty");
+    return mapper(dbSession).selectComponentsByQualifiers(qualifiers);
```
Comment (sns-seb) [inline sonar-db/src/main/java/org/sonar/db/component/ComponentDao.java]: either create a `Set` from `Collection` argument or require a `Set` as argument

File: sonar-db/src/main/java/org/sonar/db/component/ComponentDao.java [hunk 3]
```diff
@@ -251,6 +253,16 @@ public long countGhostProjects(DbSession session, @Nullable String query) {
     return mapper(dbSession).selectByProjectUuid(projectUuid);
   }
 
+  /**
+   * Retrieve enabled components keys with given qualifiers
+   *
+   * Used by Views plugin
```
Comment (sns-seb) [inline sonar-db/src/main/java/org/sonar/db/component/ComponentDao.java]: is there any specific order in the returned list?

File: sonar-db/src/main/java/org/sonar/db/component/ComponentDao.java (hunk 2 above)
Comment (julienlancelot) [inline sonar-db/src/main/java/org/sonar/db/component/ComponentDao.java]: Indeed, I can require a set as argument, and return a Set

File: sonar-db/src/main/java/org/sonar/db/component/ComponentDao.java (hunk 3 above)
Comment (julienlancelot) [inline sonar-db/src/main/java/org/sonar/db/component/ComponentDao.java]: Nope

**Comment by sns-seb** [comment]
LGTM

**Pull request merged**